File: sha256.js
// Import the crypto module
const crypto = require("crypto");

// Function to generate a SHA-256 hash
function generateSHA256Hash(data) {
  // Create a hash object, update it with data, and convert it to a hexadecimal string
  const hash = crypto.createHash("sha256").update(data).digest("hex");
  return hash;
}

// Example usage
const inputData = "Hardik";
const hashOutput = generateSHA256Hash(inputData);

console.log(`Input Data: ${inputData}`);
console.log(`SHA-256 Hash: ${hashOutput}`);

File: SHA-256_Rust/SHA-256.rs
use sha2::{Sha256, Digest};

fn main() {
    // Input data
    let input_data = "Hello, Blockchain!";

    // Create a Sha256 object
    let mut hasher = Sha256::new();

    // Update hasher with input data
    hasher.update(input_data);

    // Retrieve the result as a byte array
    let result = hasher.finalize();

    // Convert to a hexadecimal string
    let hash_hex = format!("{:x}", result);

    // Print the input and the hash
    println!("Input Data: {}", input_data);
    println!("SHA-256 Hash: {}", hash_hex);
}
